tests/AgDatabaseMove.Integration/TestAgDatabase.cs

```diff
@@ -0,0 +1,75 @@
+﻿using AgDatabaseMove.Integration.Fixtures;
+using AgDatabaseMove.SmoFacade;
+using System.Collections.Generic;
+using Xunit;
+using Smo = Microsoft.SqlServer.Management.Smo;
+
+namespace AgDatabaseMove.Integration
+{
+  public class TestAgDatabase : IClassFixture<TestAgDatabaseFixture>
+  {
+    private readonly TestAgDatabaseFixture _testAgDatabaseFixture;
+
+    public TestAgDatabase(TestAgDatabaseFixture testAgDatabaseFixture)
+    {
+      _testAgDatabaseFixture = testAgDatabaseFixture;
+    }
+
+    private AgDatabase AgDatabase => _testAgDatabaseFixture._agDatabase;
+
+    private string LoginName => _testAgDatabaseFixture._loginName;
+
+    private string LoginPassword => _testAgDatabaseFixture._loginPassword;
+
+    private string DefaultDatabase => _testAgDatabaseFixture._loginDefaultDatabase;
+
+    private void CreateNewSqlLogin(LoginProperties login)
+    {
+      AgDatabase.AddLogin(login);
+    }
+
+    private void TestLoginExistsOnAg(List<Smo.Login> createdLogins)
+    {
+      createdLogins.ForEach(Assert.NotNull);
+    }
+
+    private List<Smo.Login> GetCreatedLogins()
+    {
+      List<Smo.Login> logins = new List<Smo.Login>();
+      logins.Add(AgDatabase._listener.Primary._server.Logins[LoginName]);
+      foreach(var server in AgDatabase._listener.Secondaries)
+      {
+        logins.Add(server._server.Logins[LoginName]);
+      }

```

Comment:
The purpose of AgDatabase is to abstract the Primary vs N number of secondary nodes.
The test here would be better if it didn't take advantage of `internal` properties.  Perhaps `_listener` and `_server` should be private members

Reply:
I believe `_listener` is used in another test class. It could be made `private` instead, but that should be done in another PR. `_server` on the other hand has multiple use-cases requiring that it stays at the same visibility.